website/static/js/fangorn.js

```diff
@@ -209,30 +213,30 @@ function resolveIconView(item) {
         var template = m('span', { 'class' : iconType});
         return template;
     }
-    if (item.data.nodeType === 'smartFolder') {
-        return returnView('smartCollection');
-    }
-    if (item.data.nodeType === 'folder') {
+    if (item.data.isDashboard) {
         return returnView('collection');
     }
-    if (item.data.nodeType === 'pointer' && item.parent().data.nodeType !== 'folder') {
+    if (item.data.nodeType === 'smartFolder' || item.data.isSmartFolder) {
+        return returnView('smartCollection');
+    }
+    if ((item.data.nodeType === 'pointer' && item.parent().data.nodeType !== 'folder') || (item.data.isPointer && !item.parent().data.isFolder)) {
         return returnView('link');
     }
-    if (item.data.nodeType === 'project') {
+    if (item.data.nodeType === 'project' || item.data.isProject) {
```

Comment:
Why is the check for `isProject` necessary? Isn't that taken care of by `nodeType === 'project'`?